src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -97,7 +97,24 @@ public State(TService service, SemanticDocument document)
                     return false;
                 }
 
-                IsConstant = Document.SemanticModel.GetConstantValue(Expression, cancellationToken).HasValue;
+                if (Document.SemanticModel.GetConstantValue(Expression, cancellationToken) is { HasValue: true, Value: var value })
+                {
+                    if (_service.IsInterpolatedStringExpression(Expression) && value is string)
+                    {
+                        // Interpolated strings can have constant values, but if it's being converted to a FormattableString
+                        // or IFormattable then we cannot treat it as one
+                        var typeInfo = Document.SemanticModel.GetTypeInfo(Expression, cancellationToken);
+                        IsConstant = typeInfo.ConvertedType?.IsFormattableStringOrIFormattable() != true;
```

Comment:
this bug probably exists in a *lot* of places.  that said, it feels pretty minor, so i don't ind spot-fixes for it.

Reply:
Yep. This particular function is only used in a few places, but I figured that I may as well hit them all at once.